tests/extension/oneapi_device_global/device_global_exceptions.cpp

```diff
@@ -49,24 +49,26 @@ void run_test(util::logger& log, const std::string& type_name) {
   std::shared_ptr<T[N]> src(new T[N]);
   std::shared_ptr<T[N]> dest(new T[N]);
 
-  bool is_exception_thrown{true};
-  bool is_exception_correct{true};
+  bool is_exception_thrown{false};
+  bool is_exception_correct{false};
   auto queue = util::get_cts_object::queue();
 
   queue.submit([&](sycl::handler& cgh) {
     // Try to write data beyond the end of the destination variable
     // Have to throw an exception
     try {
       cgh.template copy<T>(src.get(), dev_global<T>, N, N / 2);
-      is_exception_thrown &= false;
+      is_exception_thrown = false;
     } catch (sycl::exception const& e) {
-      is_exception_correct &= is_errc_invalid(e);
+      is_exception_thrown = true;
+      is_exception_correct = is_errc_invalid(e);
     }
 
     try {
       cgh.template copy<T>(dev_global<T>, dest.get(), N, N / 2);
-      is_exception_thrown &= false;
+      is_exception_thrown = false;
     } catch (sycl::exception const& e) {
+      is_exception_thrown &= true;
```

Comment:
This code is not correct if the first check fails. I don't think that changing default value to `false` is a good idea.

Reply:
I don't think that code is not correct. If the first check fails and the second is not, then `is_exception_thrown` will contain false and the test will fail as it is supposed to.

But I agree that `false` as the default value is not a good idea, because it makes code more complicated. If the initial value is `false`, you have to change `is_exception_thrown` in every case, not only in failure cases.

I have changed the initial value for the result variables to `true`.

Reply:
> I don't think that code is not correct. If the first check fails and the second is not, then `is_exception_thrown` will contain false and the test will fail as it is supposed to.

Maybe I misunderstand the intention of this line. What is line of code supposed to be doing then?

Reply:
This line is supposed to set `is_exception_thrown` to `true` if an exception was thrown, but only if the first invocation of member function caused exception too. If the first invocation did not cause an exception, then `is_exception_thrown` has to be `false` until the end of the test. To implement such behavior I used operator `&=` to avoid overwriting `false` to `true`.

Anyway, I changed the initial value to `true` so it doesn't matter anymore.